### PR TITLE
subtitlecomposer: init at 0.8.0

### PR DESCRIPTION
**Pull request by kugland**
*subtitlecomposer* package

- Added myself (kugland) as maintainer
- Added subtitlecomposer package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kugland** [comment]
Whis will close #221218.

**Comment by quantenzitrone** [comment]
1. The commit message for the maintainer addition should be: `maintainers: add kugland`
2. The other 3 commits should be squashed to 1 with the commit message: `subtitlecomposer: init at 0.8.0`

other than that lgtm

**Comment by kugland** [comment]
> 1. The commit message for the maintainer addition should be: `maintainers: add kugland`
> 
>     2. The other 3 commits should be squashed to 1 with the commit message: `subtitlecomposer: init at 0.8.0`
> 
> 
> see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
> 
> other than that lgtm

done

**Review by quantenzitrone — APPROVED** [review]

**Comment by quantenzitrone** [comment]
Result of `nixpkgs-review pr 273272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>subtitlecomposer</li>
  </ul>
</details>

File: pkgs/by-name/su/subtitlecomposer/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+libsForQt5.mkDerivation rec {
+  pname = "subtitlecomposer";
+  version = "0.8.0";
+
+  src = fetchFromGitLab {
+    domain = "invent.kde.org";
+    owner = "multimedia";
+    repo = "subtitlecomposer";
+    rev = "v${version}";
+    sha256 = "sha256-RKS3VTtpxnox0hzessMHmoGPpT+Ho0b3fxtQMGw9OrM=";
```
Comment (eclairevoyant) [inline pkgs/by-name/su/subtitlecomposer/package.nix:19]:
```suggestion
    hash = "sha256-RKS3VTtpxnox0hzessMHmoGPpT+Ho0b3fxtQMGw9OrM=";
```

File: pkgs/by-name/su/subtitlecomposer/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  nativeBuildInputs = [ cmake extra-cmake-modules ];
```
Comment (eclairevoyant) [inline pkgs/by-name/su/subtitlecomposer/package.nix:22]: I'm surprised to not see `wrapQtAppsHook`

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Feedback below.

File: pkgs/by-name/su/subtitlecomposer/package.nix (hunk 2 above)
Comment (kugland) [reply pkgs/by-name/su/subtitlecomposer/package.nix:22]: Should I add it?

**Comment by kugland** [comment]
@eclairevoyant, is there stil something wrong with the PR?

**Review by quantenzitrone — APPROVED** [review]

**Comment by teto** [comment]
Result of `nixpkgs-review pr 273272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>subtitlecomposer</li>
  </ul>
</details>

**Comment by quantenzitrone** [comment]
Result of `nixpkgs-review pr 273272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>subtitlecomposer</li>
  </ul>
</details>

**Review by quantenzitrone — APPROVED** [review]

File: pkgs/by-name/su/subtitlecomposer/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitLab
+, cmake
+, extra-cmake-modules
+, ffmpeg
+, openal
+, libsForQt5
+}:
+
+libsForQt5.mkDerivation rec {
```
Comment (eclairevoyant) [inline pkgs/by-name/su/subtitlecomposer/package.nix:10]:
```suggestion
stdenv.mkDerivation rec {
```

**Review by eclairevoyant — DISMISSED** [review]
one more change from my end, then should be good to go

File: pkgs/by-name/su/subtitlecomposer/package.nix (hunk 3 above)
Comment (kugland) [reply pkgs/by-name/su/subtitlecomposer/package.nix]: Done!

File: pkgs/by-name/su/subtitlecomposer/package.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    description = "An open source text-based subtitle editor";
+    longDescription = ''
+      An open source text-based subtitle editor that supports basic and
+      advanced editing operations, aiming to become an improved version of
+      Subtitle Workshop for every platform supported by Plasma Frameworks.
+    '';
+    changelog = "https://invent.kde.org/multimedia/subtitlecomposer/-/blob/master/ChangeLog";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ kugland ];
+    mainProgram = "subtitlecomposer";
```
Comment (eclairevoyant) [inline pkgs/by-name/su/subtitlecomposer/package.nix:48]:
```suggestion
    mainProgram = "subtitlecomposer";
    platforms = with platforms; linux ++ freebsd ++ windows`
```

Oh, I missed this earlier - upstream [only builds for linux, freebsd, and windows](https://invent.kde.org/multimedia/subtitlecomposer/-/blob/master/.gitlab-ci.yml?ref_type=heads)

i.e. `meta.platforms` should be set accordingly here

Sorry for the extra churn :grimacing:

**Pull request merged**